src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -232,141 +261,137 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
-    auto & groupList  = commandData.groupList;
-
-    EmberStatus status = EMBER_ZCL_STATUS_FAILURE;
-    uint8_t i, j;
-    uint8_t count = 0;
-    uint8_t list[EMBER_BINDING_TABLE_SIZE << 1];
-    uint8_t listLen = 0;
-    CHIP_ERROR err  = CHIP_NO_ERROR;
-
-    emberAfGroupsClusterPrint("RX: GetGroupMembership [");
-
-    // For all networks, Get Group Membership commands may be sent either
-    // unicast or broadcast (removing the ZLL-specific limitation to unicast).
-
-    // When Group Count is zero, respond with a list of all active groups.
-    // Otherwise, respond with a list of matches.
-    // TODO: https://github.com/project-chip/connectedhomeip/issues/10335
-    if (groupCount == 0)
+public:
+    // Use GetCommandId instead of commandId directly to avoid naming conflict with CommandIdentification in ExecutionOfACommand
+    static constexpr CommandId GetCommandId() { return Commands::GetGroupMembershipResponse::Id; }
+    static constexpr ClusterId GetClusterId() { return Groups::Id; }
+
+    GetGroupMembershipResponse(const Commands::GetGroupMembership::DecodableType & data,
+                               GroupDataProvider::GroupMappingIterator * iter) :
+        mCommandData(data),
+        mIterator(iter)
+    {}
+
+    const Commands::GetGroupMembership::DecodableType & mCommandData;
+    GroupDataProvider::GroupMappingIterator * mIterator = nullptr;
+    // A capacity of 0xFF means that it is unknown if any further groups may be
+    // added.  Each group requires a binding and, because the binding table is
+    // used for other purposes besides groups, we can't be sure what the
+    // capacity will be in the future.
+    uint8_t mCapacity = 0xff;
+
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, TLV::Tag tag) const
     {
-        for (i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+        TLV::TLVType outer;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::ContextTag(to_underlying(chip::app::CommandDataIB::Tag::kData)),
+                                                   TLV::kTLVType_Structure, outer));
+        ReturnErrorOnFailure(chip::app::DataModel::Encode(
+            writer, TLV::ContextTag(to_underlying(Groups::Commands::GetGroupMembershipResponse::Fields::kCapacity)), 0xff));
         {
-            EmberBindingTableEntry entry;
-            status = emberGetBinding(i, &entry);
-            if ((status == EMBER_SUCCESS) && (entry.type == EMBER_MULTICAST_BINDING) && (entry.local == emberAfCurrentEndpoint()))
+            TLV::TLVType type;
+            ReturnErrorOnFailure(writer.StartContainer(
+                TLV::ContextTag(to_underlying(Groups::Commands::GetGroupMembershipResponse::Fields::kGroupList)),
+                TLV::kTLVType_Array, type));
+
+            GroupDataProvider::GroupMapping mapping;
+            size_t requestCount = 0;
+            size_t matchCount   = 0;
+            ReturnErrorOnFailure(mCommandData.groupList.ComputeSize(&requestCount));
```

Comment:
If `requestCount` is 0, we should just return all the results from mIterator, right?  That codepath seems to have gone missing....

Reply:
Correct. I added the missing condition, and tried to add a YAML test to check this condition, unfortunately the code generation fails when the argument is an empty list.

Reply:
Yeah.  https://github.com/project-chip/connectedhomeip/pull/11496 fixes, but fine to land this without that test and then we can add it once both PRs are in.